client/web/admin/src/views/System/Template/Editor.vue

```diff
@@ -131,18 +133,28 @@ export default {
           this.fetchTemplate()
         } else {
           this.template = new system.Template()
+          this.initialTemplateState = new system.Template()
         }
       },
     },
   },
 
+  beforeRouteUpdate (to, from, next) {
+    this.checkUnsavedChanges(next)
+  },
+
+  beforeRouteLeave (to, from, next) {
+    this.checkUnsavedChanges(next)
+  },
+
   methods: {
     fetchTemplate () {
       this.incLoader()
 
       this.$SystemAPI.templateRead({ templateID: this.templateID })
         .then(t => {
           this.template = new system.Template(t)
+          this.initialTemplateState = new system.Template(cloneDeep(t))
```

Comment:
probably makes sense to implement .clone for all resources, so you can freely just use this.template.clone(). Same for all other resources that have a class/type.

If they dont have it this approach is fine